### PR TITLE
Drop support php 7, add php 8.2 support

### PR DESCRIPTION
**Pull request by fezfez**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -5,15 +5,12 @@
     "license": "BSD-3-Clause",
     "config": {
         "sort-packages": true,
-        "platform": {
-            "php": "7.4.99"
-        },
```
Comment (Ocramius) [inline composer.json]: Should be kept, but updated

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (fezfez) [reply composer.json]: 8.0.99 ?

Comment (Ocramius) [reply composer.json]: yup

This prevents incompatible dependency auto-upgrades

Comment (Ocramius) [reply composer.json]: @fezfez was the lockfile updated?

Comment (fezfez) [reply composer.json]: > @fezfez was the lockfile updated?

yep

**Review by Ocramius — CHANGES_REQUESTED** [review]
@fezfez it states:


```
 Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. It is recommended that you run `composer update` or `composer update <package name>`.
```

File: composer.json (hunk 1 above)
Comment (fezfez) [reply composer.json]: > @fezfez it states:
> 
> ```
>  Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. It is recommended that you run `composer update` or `composer update <package name>`.
> ```

done

**Review by Ocramius — APPROVED** [review]
Thanks @fezfez!

**Pull request merged**